service/mailgun/mailgun.go

```diff
@@ -0,0 +1,53 @@
+package mailgun
+
+import (
+	"context"
+
+	"github.com/mailgun/mailgun-go/v4"
+	"github.com/pkg/errors"
+)
+
+// Mailgun struct holds necessary data to communicate with the Mailgun API.
+type Mailgun struct {
+	client            mailgun.Mailgun
+	senderAddress     string
+	receiverAddresses []string
+}
+
+// New returns a new instance of a Mailgun notification service.
+// You will need a Mailgun API key and domain name.
+// See https://documentation.mailgun.com/en/latest/
+func New(domain, apiKey, senderAddress string, opts ...Option) *Mailgun {
+	client := mailgun.NewMailgun(domain, apiKey)
+
+	m := &Mailgun{
+		client:            client,
+		senderAddress:     senderAddress,
+		receiverAddresses: []string{},
+	}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m
+}
+
+// AddReceivers takes email addresses and adds them to the internal address list. The Send method will send
+// a given message to all those addresses.
+func (m *Mailgun) AddReceivers(addresses ...string) {
+	m.receiverAddresses = append(m.receiverAddresses, addresses...)
+}
+
+// Send takes a message subject and a message body and sends them to all previously set chats. Message body supports
+// html as markup language.
+func (m Mailgun) Send(subject, message string) error {
+	mailMessage := m.client.NewMessage(m.senderAddress, subject, message, m.receiverAddresses...)
+
+	_, _, err := m.client.Send(context.Background(), mailMessage)
+	if err != nil {
```

Comment:
Could combile these two lines:
  `if _, _, err := m.client.Send(context.Background(), mailMessage); err != nil {
		return errors.Wrap(err, "failed to send mail using Mailgun service")
	}`

Reply:
Combined these two lines on a new commit

Reply:
@NotVinay appreciate your review and your of course right with your suggestion but since this is a question of style and not function and the rest of the code follows the guideline to split the error assignment and evaluation I would want you @chanioxaris to please revert this to how you originally did it.

Reply:
Reverted the change on new commit

Reply:
Thanks!